selfdrive/sensord/sensors_qcom2.cc

```diff
@@ -179,33 +169,30 @@ int sensor_loop() {
   util::set_core_affinity({1});
   std::system("sudo su -c 'echo 1 > /proc/irq/336/smp_affinity_list'");
 
-  PubMaster pm({"sensorEvents"});
+  PubMaster pm({"gyroscope", "gyroscope2", "accelerometer", "accelerometer2",
+                "temperatureSensor", "lightSensor", "magnetometer"});
   init_ts = nanos_since_boot();
 
   // thread for reading events via interrupts
   std::vector<Sensor *> lsm_interrupt_sensors = {&lsm6ds3_accel, &lsm6ds3_gyro};
-  std::thread lsm_interrupt_thread(&interrupt_loop, lsm6ds3_accel.gpio_fd, std::ref(lsm_interrupt_sensors), std::ref(pm));
+  std::thread lsm_interrupt_thread(&interrupt_loop, std::ref(lsm_interrupt_sensors),
+                                   std::ref(pm), std::ref(sensor_service));
 
   // polling loop for non interrupt handled sensors
   while (!do_exit) {
     std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
 
-    const int num_events = sensors.size();
-    MessageBuilder msg;
-    auto sensor_events = msg.initEvent().initSensorEvents(num_events);
-
-    for (int i = 0; i < num_events; i++) {
-      auto event = sensor_events[i];
-      sensors[i]->get_event(event);
-    }
+    for (Sensor *sensor : sensors) {
+      MessageBuilder msg;
+      if (!sensor->get_event(msg)) {
+        continue;
+      }
 
-    if (nanos_since_boot() - init_ts < init_delay) {
-      continue;
-    }
+      if (!sensor->is_data_valid(init_ts, nanos_since_boot())) {
+        continue;
+      }
```

Comment:
check before getting the event

Reply:
I think it makes sense to do it after, just to have the full communication with the sensor